exif.cpp

```diff
@@ -380,7 +380,7 @@ IFEntry parseIFEntry_temp(const unsigned char *buf, const unsigned offs,
       }
       // and cut zero byte at the end, since we don't want that in the
       // std::string
-      if (result.length() &&
+      if (!result.val_string().empty() &&
```

Comment:
What's the reason for doing this?

It doesn't add any clarity, and `length()` isn't accessing a string - it's accessing `length_` which is an unsigned int.

Reply:
clang-tidy gripes about using an integer value in a boolean expression. If an `empty()` interface is available, it recommends to use that instead of using something like `length()` or `size()` as a boolean.

It's entirely stylistic (and in my opinion `!empty()` is actually less clear), but linting tools really like to enforce this style.

Reply:
If `result` were a string or container, then changing to `empty()` is fine, but we already have the length stored separately from the data. In this case using `length()` is just clearer to me than using the convoluted empty string check.

If it's complaining about using an integer value in a boolean expression, then I would change it to:

```cpp
(result.length() > 0)
```